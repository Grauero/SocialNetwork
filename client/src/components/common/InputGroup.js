import React from 'react';
import propTypes from 'prop-types';
import classnames from 'classnames';

const InputGroup = ({
  name,
  placeholder,
  value,
  error,
  onChange,
  icon,
  type
}) => (
  <div className="input-group mb-3">
    <div className="input-group-prepend">
      <span className="input-group-text">
        <i className={icon} />
      </span>
    </div>
    <input
      type={type}
      className={classnames('form-control form-control-lg', {
        'is-invalid': error
      })}
      placeholder={placeholder}
      name={name}
      value={value}
      onChange={onChange}
    />
    {error && <div className="invalid-feedback">{error}</div>}
  </div>
);

InputGroup.propTypes = {
  name: propTypes.string.isRequired,
  placeholder: propTypes.string,
  value: propTypes.string.isRequired,
  icon: propTypes.string,
  type: propTypes.string,
  error: propTypes.string,
  onChange: propTypes.func.isRequired
};

InputGroup.defaultProps = {
  placeholder: null,
  icon: null,
  error: null,
  type: 'text'
};

export default InputGroup;
