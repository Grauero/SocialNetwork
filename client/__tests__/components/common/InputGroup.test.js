import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import InputGroup from '../../../src/components/common/InputGroup';

const props = {
  name: 'name',
  value: 'value',
  type: 'text',
  onChange: jest.fn()
};
let component;

beforeEach(() => (component = mount(<InputGroup {...props} />)));

afterEach(() => component.unmount());

it('renders correct input with provided props', () => {
  const input = component.find('input');
  const type = input.getDOMNode().getAttribute('type');
  const name = input.getDOMNode().getAttribute('name');
  const value = input.getDOMNode().getAttribute('value');

  expect(type).toBe(props.type);
  expect(name).toBe(props.name);
  expect(value).toBe(props.value);
});

it('handles input change', () => {
  component.find('input').simulate('change');

  expect(props.onChange).toHaveBeenCalled();
  expect(props.onChange).toHaveBeenCalledTimes(1);
});

it('renders error if error is provided with props', () => {
  const propsWithError = Object.assign({}, props, { error: 'Error' });
  const component = mount(<InputGroup {...propsWithError} />);
  const errorDiv = component.find('.invalid-feedback');
  const errorClass = errorDiv.getDOMNode().getAttribute('class');

  expect(errorDiv.debug()).toBeTruthy();
  expect(errorClass).toBe('invalid-feedback');
});

it('matches snapshot', () => {
  expect(toJSON(component)).toMatchSnapshot();
});
