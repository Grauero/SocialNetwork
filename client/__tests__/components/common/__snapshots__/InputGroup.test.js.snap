// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<InputGroup
  error={null}
  icon={null}
  name="name"
  onChange={[Function]}
  placeholder={null}
  type="text"
  value="value"
>
  <div
    className="input-group mb-3"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
      >
        <i
          className={null}
        />
      </span>
    </div>
    <input
      className="form-control form-control-lg"
      name="name"
      onChange={[Function]}
      placeholder={null}
      type="text"
      value="value"
    />
  </div>
</InputGroup>
`;
